borg/archiver.py

```diff
@@ -286,13 +292,25 @@ def do_extract(self, args):
         manifest, key = Manifest.load(repository)
         archive = Archive(repository, key, manifest, args.location.archive,
                           numeric_owner=args.numeric_owner)
-        patterns = adjust_patterns(args.paths, args.excludes)
+
+        matcher = PatternMatcher()
+        if args.excludes:
+            matcher.add(args.excludes, False)
```

Comment:
I would rather not add include/exclude-specific code to `PatternMatcher`. If you look at the two uses they're a bit different. Keeping the class agnostic makes testing much easier. An earlier version had `add_include` and `add_exclude`, but that didn't work with the way the return values are used for filtering.